src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -111,7 +111,12 @@ protected override TDeclarationNode AddField<TDeclarationNode>(TDeclarationNode
 
         protected override TDeclarationNode AddMethod<TDeclarationNode>(TDeclarationNode destination, IMethodSymbol method, CodeGenerationOptions options, IList<bool> availableIndices)
         {
-            CheckDeclarationNode<TypeDeclarationSyntax, CompilationUnitSyntax, NamespaceDeclarationSyntax>(destination);
+            if (destination is PropertyDeclarationSyntax || destination is IndexerDeclarationSyntax)
+            {
+                return destination;
+            }
+
+            CheckDeclarationNode<TypeDeclarationSyntax, CompilationUnitSyntax, NamespaceDeclarationSyntax, MethodDeclarationSyntax, LocalFunctionStatementSyntax>(destination);
```

Comment:
this doesn't seem right.  you shouldn't be able to AddMethod to anything but a `TypeDeclarationSyntax, CompilationUnitSyntax, NamespaceDeclarationSyntax`.  I also don't think we particularly need an AddLocalFunction helper.  It's just a statement, so it should be easy to add to the method we're in.